Sources/momento/CacheClient.swift

```diff
@@ -176,4 +177,423 @@ public class CacheClient: CacheClientProtocol {
         }
         return await self.dataClient.set(cacheName: cacheName, key: key, value: value, ttl: ttl)
     }
+    
+    /**
+     Adds multiple elements to the back of the given list. Creates the list if it does not already exist.
+     - Parameters:
+        - cacheName: the name of the cache to store the list in
+        - listName: the list to add to
+        - values: the elements to add to the list
+        - truncateFrontToSize: If the list exceeds this length, remove excess from the front of the list. Must be positive.
+        - ttl: refreshes the list's TTL using the client's default if this is not supplied.
+     - Returns: CacheListConcatenateBackResponse representing the result of the operation.
+     Pattern matching can be used to operate on the appropriate subtype.
+    ```
+     switch response {
+     case let responseError as CacheListConcatenateBackError:
+        // handle error
+     case let responseSuccess as CacheListConcatenateBackSuccess:
+        // handle success
+     }
+    ```
+     */
+    public func listConcatenateBack(
+        cacheName: String,
+        listName: String,
+        values: [ScalarType],
+        truncateFrontToSize: Int? = nil,
+        ttl: CollectionTtl? = nil
+    ) async -> CacheListConcatenateBackResponse {
+        do {
+            try validateCacheName(cacheName: cacheName)
+            try validateListName(listName: listName)
+            try validateTruncateSize(size: truncateFrontToSize)
```

Comment:
We do also need to validate the `CollectionTtl` here to make sure the `ttl` property is > 0. It looks like this is inconsistently implemented across other SDKs but there is an example in the [go sdk](https://github.com/momentohq/client-sdk-go/blob/b5e716f3e26b01b9031b004b75852aba7e3a8288/momento/requester.go#L172). That method does some additional prep that you're handling in the `DataClient` already, but I would like to do a quick check here that `CollectionTtl.ttl` is > 0 and return an `InvalidArgument` error if it isn't.